File: go/fixture/go-mini/internal/models/priority.go
package models

import "fmt"

// ParsePriority accepts the integer devices send on the wire and rejects
// anything outside the range the scheduler knows how to order.
func ParsePriority(n int) (Priority, error) {
	if n < int(Low) || n > int(High) {
		return 0, fmt.Errorf("priority %d: want %d-%d", n, Low, High)
	}
	return Priority(n), nil
}

// Outranks reports whether p should be scheduled ahead of o.
func (p Priority) Outranks(o Priority) bool {
	return p > o
}

File: py/fixture/py-mini/internal/models/priority.py
"""Parsing of the priority devices send on the wire."""

from internal.models.job_kind import Priority


def parse_priority(n: int) -> Priority:
    """Accept the integer devices send on the wire.

    Anything outside the range the scheduler knows how to order is rejected.
    """
    if n < int(Priority.LOW) or n > int(Priority.HIGH):
        raise ValueError(f"priority {n}: want {int(Priority.LOW)}-{int(Priority.HIGH)}")
    return Priority(n)
